solr/solrj/src/java/org/apache/solr/client/solrj/impl/HttpSolrClientBuilderBase.java

```diff
@@ -0,0 +1,193 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.solr.client.solrj.impl;
+
+import java.net.CookieStore;
+import java.util.Set;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
+import org.apache.solr.client.solrj.ResponseParser;
+import org.apache.solr.client.solrj.request.RequestWriter;
+
+public abstract class HttpSolrClientBuilderBase {
+  protected Long idleTimeoutMillis;
+  protected Long connectionTimeoutMillis;
+  protected Long requestTimeoutMillis;
+  protected String basicAuthAuthorizationStr;
+  protected Boolean followRedirects;
+  protected String baseSolrUrl;
+  protected RequestWriter requestWriter;
+  protected ResponseParser responseParser;
+  protected String defaultCollection;
+  protected Set<String> urlParamNames;
+  protected Integer maxConnectionsPerHost;
+  protected ExecutorService executor;
+  protected CookieStore cookieStore;
+  protected String proxyHost;
+  protected int proxyPort;
+  protected boolean proxyIsSocks4;
+  protected boolean proxyIsSecure;
+
+  public HttpSolrClientBuilderBase() {}
+
+  protected abstract <B extends Http2SolrClientBase> B build(Class<B> type);
```

Comment:
I think "C" would be a better type variable name because this *is* the client.  In another comment you'll see I suggest adding another type variable for the Builder, which would naturally be "B".